packages/gateway/src/manager.ts

```diff
@@ -54,9 +55,165 @@ export function createGatewayManager(options: CreateGatewayManagerOptions): Gate
+
+        // Resetting buckets
+        gateway.buckets.clear()
+        // Refilling buckets with new values
+        gateway.prepareBuckets()
+
+        // SPREAD THIS OUT TO DIFFERENT WORKERS TO BEGIN STARTING UP
+        gateway.buckets.forEach(async (bucket, bucketId) => {
+          for (const worker of bucket.workers) {
+            for (const shardId of worker.queue) {
+              await gateway.resharding.tellWorkerToPrepare(worker.id, shardId, bucketId)
+            }
+          }
+        })
+      },
+      async tellWorkerToPrepare(workerId, shardId, bucketId) {
+        logger.warn(`[Resharding] Telling worker to prepare. Worker: ${workerId} | Shard: ${shardId} | Bucket: ${bucketId}`)
+        const shard = new Shard({
+          id: shardId,
+          connection: {
+            compress: gateway.compress,
+            intents: gateway.intents,
+            properties: gateway.properties,
+            token: gateway.token,
+            totalShards: gateway.totalShards,
+            url: gateway.url,
+            version: gateway.version,
+          },
+          events: {},
+          requestIdentify: async () => {
+            await gateway.identify(shardId)
+          },
+          shardIsReady: async () => {
+            logger.debug(`[Shard] Shard #${shardId} is ready`)
+            await delay(gateway.spawnShardDelay)
+            logger.debug(`[Shard] Resolving shard identify request`)
+            gateway.buckets.get(shardId % gateway.connection.sessionStartLimit.maxConcurrency)!.identifyRequests.shift()?.()
+          },
+        })
+
+        if (gateway.preferSnakeCase) {
+          shard.forwardToBot = async (payload) => {
+            if (payload.t === 'READY') {
+              await gateway.resharding.updateGuildsShardId(
+                (payload.d as DiscordReady).guilds.map((g) => g.id),
+                shardId,
+              )
+            }
+
+            options.events?.message?.(shard, payload)
+          }
+        }
+
+        gateway.resharding.shards.set(shardId, shard)
+
+        const bucket = gateway.buckets.get(shardId % gateway.connection.sessionStartLimit.maxConcurrency)
+        if (!bucket) return
+
+        return await new Promise((resolve) => {
+          // Mark that we are making an identify request so another is not made.
+          bucket.identifyRequests.push(resolve)
+          logger.debug(`[Gateway] identifying shard #(${shardId}).`)
+          // This will trigger identify and when READY is received it will resolve the above request.
+          shard?.identify().then(async () => {
+            // Tell the manager that this shard is online
+            return await gateway.resharding.shardIsPending(shard)
+          })
+        })
+      },
+      async updateGuildsShardId(guildIds, shardId) {
+        logger.debug(
+          `[Resharding] Updating the following guild ids shard to #${shardId}: ${guildIds.join(', ')}. Override this function to update your cache if you need to.`,
```

Comment:
Small thing, i don't think it should be the logger informing the user about the possibility to override "this function". Also because there is no way (without searching in discordeno code) to know what "this function" refers to when it is in the console

Reply:
I'll be covering these in big bot guide under resharding section so that should be good enough I guess